### PR TITLE
Swerve changes

### PR DESCRIPTION
**Pull request by Irishumanoid**
Competition code

### CODE REVIEW

File: src/main/java/frc/robot/Commands/ArmTest.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ArmTest.java]: Please remove this file

File: src/main/java/frc/robot/Commands/ElevatorCommand.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ElevatorCommand.java:1]: Can we stub out this file (leave the methods empty)? I'd rather us start it from a blank slate.

File: src/main/java/frc/robot/Commands/ElevatorPIDCommand.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ElevatorPIDCommand.java]: Let's delete this file

File: src/main/java/frc/robot/Commands/IntakeCommand.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/IntakeCommand.java]: nit: can we adjust this comment?

File: src/main/java/frc/robot/Commands/IntakeCommand.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/IntakeCommand.java]: nit: we should probably remove this debug statement, right?

File: src/main/java/frc/robot/Commands/ReleaseCommand.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ReleaseCommand.java]: nit: adjust this comment

File: src/main/java/frc/robot/Commands/ScoreGamePieceCommand.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ScoreGamePieceCommand.java:1]: Can we stub this out please?

File: src/main/java/frc/robot/Commands/SimpleDriveCommand.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/SimpleDriveCommand.java]: nit: let's use the values from Constants.java here instead of the hard-coded values

File: src/main/java/frc/robot/Commands/ToStartConfigCommand.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ToStartConfigCommand.java:9]: Do we need this? If so, let's stub it out.

File: src/main/java/frc/robot/Constants.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Constants.java:60]: Just curious, when did these change?

File: src/main/java/frc/robot/Constants.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Constants.java:157]: These aren't the final values right? Let's mark them with TODO comments. Also, it'd be nice if we could determine the units for these (i.e., meters, inches, furlongs, etc.)

File: src/main/java/frc/robot/Constants.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Constants.java]: These ports are correct, right? Let's remove this comment :)

File: src/main/java/frc/robot/Constants.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Constants.java:172]: Do we need these?

File: src/main/java/frc/robot/Constants.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Constants.java:179]: What are these for?

File: src/main/java/frc/robot/Constants.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Constants.java:188]: nit: enum values should be on separate lines

File: src/main/java/frc/robot/GenericPID/Approximations/ApproximateDerivative.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/GenericPID/Approximations/ApproximateDerivative.java]: Delete this please.

File: src/main/java/frc/robot/GenericPID/Approximations/ApproximateIntegral.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/GenericPID/Approximations/ApproximateIntegral.java]: Delete this please

File: src/main/java/frc/robot/GenericPID/Approximations/DoubleFunction.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/GenericPID/Approximations/DoubleFunction.java]: Delete

File: src/main/java/frc/robot/GenericPID/Approximations/Zero.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/GenericPID/Approximations/Zero.java]: Delete

File: src/main/java/frc/robot/GenericPID/GenericPID.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/GenericPID/GenericPID.java]: Delete

File: src/main/java/frc/robot/GenericPID/PIDConfig.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/GenericPID/PIDConfig.java]: Delete

File: src/main/java/frc/robot/GenericPID/Testing/Motor.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/GenericPID/Testing/Motor.java]: Delete

File: src/main/java/frc/robot/Logger/Logger.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Logger/Logger.java]: What is the purpose of this class?

File: src/main/java/frc/robot/Logger/LoggerEntryBase.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Logger/LoggerEntryBase.java]: Is this in use?

File: src/main/java/frc/robot/Logger/LoggerRawEntryBase.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Logger/LoggerRawEntryBase.java]: In use?

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java]: nit: what is this comment for?

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java:75]: Can we remove unused autos for now and stub them out?

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java]: Remove dead code please

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java:181]: Tag with TODO please

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java:192]: Tag with TODO

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java:245]: TODO

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java:272]: TODO

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java:348]: nit: enum entries should be on different lines

File: src/main/java/frc/robot/PathPlanningCode/AutoUtils.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/AutoUtils.java]: nit: different lines

File: src/main/java/frc/robot/PathPlanningCode/GenerateTrajectoryFromFile.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/PathPlanningCode/GenerateTrajectoryFromFile.java:14]: What does this do and does it work?

File: src/main/java/frc/robot/RobotContainer.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/RobotContainer.java]: Remove dead code

File: src/main/java/frc/robot/RobotContainer.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/RobotContainer.java]: nit: add some comments

File: src/main/java/frc/robot/RobotContainer.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/RobotContainer.java:117]: There's other ways of doing this right? This seems a bit odd.

File: src/main/java/frc/robot/RobotTest.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/RobotTest.java]: Please delete this class -- it shouldn't be used.

File: src/main/java/frc/robot/Subsystems/ArmSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/ArmSubsystem.java]: Why did we add this?

File: src/main/java/frc/robot/Subsystems/DriveSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/DriveSubsystem.java]: Is this in use?

File: src/main/java/frc/robot/Subsystems/DriveSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/DriveSubsystem.java]: Delete this comment -- it's not meaningful

File: src/main/java/frc/robot/Subsystems/DriveSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/DriveSubsystem.java]: delete this comment -- it's not meaningful

File: src/main/java/frc/robot/Subsystems/DriveSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/DriveSubsystem.java]: Delete

File: src/main/java/frc/robot/Subsystems/DriveSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/DriveSubsystem.java]: If we think this is useful for diagnostics (it could be) then let's uncomment it. Otherwise, let's delete it.

File: src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java]: nit: TODO (instead of "to do")

File: src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java:31]: Let's clean this up before we merge. Remove all unused variables, constants, etc.

File: src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java]: Dead code

File: src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java]: If we think this is useful for future diagnostics let's put it on SmartDashboard, otherwise we should remove it.

File: src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java]: Do we need this?

File: src/main/java/frc/robot/Subsystems/ElevatorTrapezoidalSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/ElevatorTrapezoidalSubsystem.java]: Delete this class please.

File: src/main/java/frc/robot/Subsystems/VisionSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/VisionSubsystem.java:48]: nit: at some point we should re-name this

File: src/main/java/frc/robot/Subsystems/VisionSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/VisionSubsystem.java:35]: What are these for?

File: src/main/java/frc/robot/Subsystems/VisionSubsystem.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/VisionSubsystem.java]: nit: "i" isn't a particularly descriptive loop variable since we're using a for-each loop ("enhanced for loop") -- I'd suggest something more like "target"

**Review by garrettwitowski — CHANGES_REQUESTED** [review]
Overall pretty good. To get approval, please:
1. Remove dead/unused code
2. Clean up the ElevatorSubsystem
3. Create a backlog item to track the other comments/nits

File: src/main/java/frc/robot/Constants.java
Comment (Irishumanoid) [reply src/main/java/frc/robot/Constants.java:60]: when we switched to the competition chassis

File: src/main/java/frc/robot/Constants.java
Comment (Irishumanoid) [reply src/main/java/frc/robot/Constants.java:157]: they're in meters

File: src/main/java/frc/robot/Constants.java
Comment (Irishumanoid) [reply src/main/java/frc/robot/Constants.java:172]: once we implement the trapezoidal profile

File: src/main/java/frc/robot/Constants.java
Comment (Irishumanoid) [reply src/main/java/frc/robot/Constants.java:179]: for arm

File: src/main/java/frc/robot/Subsystems/VisionSubsystem.java
Comment (Irishumanoid) [reply src/main/java/frc/robot/Subsystems/VisionSubsystem.java:35]: adding april tag positions manually

File: src/main/java/frc/robot/RobotContainer.java
Comment (Irishumanoid) [reply src/main/java/frc/robot/RobotContainer.java:117]: yes but it works for now and I don't want to review this pr forever

**Review by Irishumanoid — COMMENTED** [review]
made most requested edits

**Review by garrettwitowski — APPROVED** [review]

**Pull request merged**
